--- configure.ac.orig	2011-01-02 03:53:43.765000111 +0200
+++ configure.ac	2011-01-02 04:01:16.325000112 +0200
@@ -117,10 +117,26 @@
 PKG_CHECK_MODULES(EXIV2, exiv2 >= 0.18)
 PKG_CHECK_MODULES(PNG, libpng >= 1.2)
 PKG_CHECK_MODULES(OPENEXR, OpenEXR >= 1.6)
-PKG_CHECK_MODULES(GPHOTO2, [libgphoto2 >= 2.4.5])
+#PKG_CHECK_MODULES(GPHOTO2, [libgphoto2 >= 2.4.5])
 PKG_CHECK_MODULES(CURL, [libcurl >= 7.18.0])
 PKG_CHECK_MODULES(DBUSGLIB, [dbus-glib-1 >= 0.80])
 
+AC_ARG_ENABLE(gphoto2,
+	      AC_HELP_STRING([--enable-gphoto2], [build with gphoto2 camera support]),
+	[
+	  if test "$enable_gphoto2" = "yes"; then
+	    PKG_CHECK_MODULES(GPHOTO2, [libgphoto2 >= 2.4.5])
+	    AC_DEFINE(HAVE_GPHOTO2, 1, [whether build gphoto2 camera support])
+	  fi
+	  AM_CONDITIONAL([HAVE_GPHOTO2], [test "$enable_ghoto2" = "yes"])
+        ],
+	[
+          PKG_CHECK_MODULES(GPHOTO2, [libgphoto2 >= 2.4.5])
+          AC_DEFINE(HAVE_GPHOTO2, 1, [whether build gphoto2 camera support])
+          AM_CONDITIONAL([HAVE_GPHOTO2], [true])
+        ]
+)
+
 AC_ARG_ENABLE(gkeyring,
 	      AC_HELP_STRING([--enable-gkeyring], [build the gnome keyring pw storage backend]),
         [
